### PR TITLE
Refactor upgrade docs

### PR DESCRIPTION
**Pull request by csplinter**
- Added a new Upgrade Overview page
- Added high-level content for separate management / workload component upgrades
- Added mention of in-place upgrades for bare metal
- Consolidated content that was reused across bare metal and vsphere upgrade docs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/clustermgmt/cluster-upgrades/upgrade-overview.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+### Upgrading EKS Anywhere Versions
+
+Each EKS Anywhere version includes all components required to create and manage EKS Anywhere clusters. For example, this includes:
+
+- Administrative / CLI components (eksctl CLI, image-builder, diagnostics-collector)
+- Management components (Cluster API controller, EKS Anywhere controller, provider-specific controllers)
+- Workload components (Kubernetes, Cilium)
+
+You can find details about each EKS Anywhere releases in the EKS Anywhere release manifest. The release manifest contains references to the corresponding bundle manifest for each EKS Anywhere version. Within the bundle manifest, you will find the components included in a specific EKS Anywhere version. The images running in your deployment use the same uri values specified in the bundle manifest for that component. For example, see the [bundle manifest](https://anywhere-assets.eks.amazonaws.com/releases/bundles/59/manifest.yaml) for EKS Anywhere v0.18.7.
```
Comment (taneyland) [inline docs/content/en/docs/clustermgmt/cluster-upgrades/upgrade-overview.md:38]:
```suggestion
You can find details about each EKS Anywhere releases in the EKS Anywhere release manifest. The release manifest contains references to the corresponding bundle manifest for each EKS Anywhere version. Within the bundle manifest, you will find the components included in a specific EKS Anywhere version. The images running in your deployment use the same URI values specified in the bundle manifest for that component. For example, see the [bundle manifest](https://anywhere-assets.eks.amazonaws.com/releases/bundles/59/manifest.yaml) for EKS Anywhere v0.18.7.
```

File: docs/content/en/docs/clustermgmt/cluster-upgrades/vsphere-and-cloudstack-upgrades.md [hunk 2]
```diff
@@ -6,30 +6,28 @@ aliases:
...
-Therefore, workload clusters are no longer affected by management cluster upgrades.
-Due to this change, each cluster must be individually upgraded to enjoy the latest features on all clusters.
-If you have a management cluster running EKS Anywhere version v0.15, you can successfully upgrade to EKS Anywhere version v0.16 and observe no changes to any of its workload clusters.
+- **Upgrades should never be run from ephemeral nodes (short-lived systems that spin up and down on a regular basis). It is highly recommended to run the `upgrade` command with the `--no-timeouts` option when the command is executed through automation. This prevents the CLI from timing out and enables cluster operators to fix issues preventing the upgrade from completing while the process is running. If an `eksctl anywhere` version is older than `v0.18.0` and upgrade fails, _you must not delete the KinD bootstrap cluster Docker container_. During an upgrade, the bootstrap cluster contains critical EKS Anywhere components. If it is deleted after a failed upgrade, they cannot be recovered.**
```
Comment (taneyland) [inline docs/content/en/docs/clustermgmt/cluster-upgrades/vsphere-and-cloudstack-upgrades.md:20]: Same comment as above about separating these

File: docs/content/en/docs/clustermgmt/cluster-upgrades/vsphere-and-cloudstack-upgrades.md [hunk 3]
```diff
@@ -6,30 +6,28 @@ aliases:
...
+- In EKS Anywhere version `v0.13.0`, we introduced the EKS Anywhere lifecycle controller that runs on management clusters and manages workload clusters. The EKS Anywhere lifecycle controller enables you to use Kubernetes API-compatible clients such as `kubectl`, GitOps, or Terraform for managing workload clusters. In this EKS Anywhere version, the EKS Anywhere lifecycle controller rolls out new nodes in workload clusters when management clusters are upgraded.
+- In `eksctl anywhere` version `v0.16.0`, this behavior was changed such that management clusters can be upgraded separately from workload clusters.
```
Comment (taneyland) [inline docs/content/en/docs/clustermgmt/cluster-upgrades/vsphere-and-cloudstack-upgrades.md:22]: Same comment about consistency in how we reference the EKS-A version

File: docs/content/en/docs/concepts/support-versions.md [hunk 4]
```diff
@@ -37,7 +37,7 @@ Each EKS Anywhere version includes all components required to create and manage
...
+You can find details about each EKS Anywhere releases in the [EKS Anywhere release manifest.](https://anywhere-assets.eks.amazonaws.com/releases/eks-a/manifest.yaml) The release manifest contains references to the corresponding bundle manifest for each EKS Anywhere version. Within the bundle manifest, you will find the components included in a specific EKS Anywhere version. The images running in your deployment use the same uri values specified in the bundle manifest for that component. For example, see the [bundle manifest](https://anywhere-assets.eks.amazonaws.com/releases/bundles/59/manifest.yaml) for EKS Anywhere v0.18.7.
```
Comment (taneyland) [inline docs/content/en/docs/concepts/support-versions.md:40]:
```suggestion
You can find details about each EKS Anywhere releases in the [EKS Anywhere release manifest.](https://anywhere-assets.eks.amazonaws.com/releases/eks-a/manifest.yaml) The release manifest contains references to the corresponding bundle manifest for each EKS Anywhere version. Within the bundle manifest, you will find the components included in a specific EKS Anywhere version. The images running in your deployment use the same URI values specified in the bundle manifest for that component. For example, see the [bundle manifest](https://anywhere-assets.eks.amazonaws.com/releases/bundles/59/manifest.yaml) for EKS Anywhere v0.18.7.
```

File: docs/content/en/docs/clustermgmt/cluster-upgrades/baremetal-upgrades.md [hunk 5]
```diff
@@ -6,96 +6,35 @@ aliases:
...
+- In EKS Anywhere version `v0.15.0`, we introduced the EKS Anywhere lifecycle controller that runs on management clusters and manages workload clusters. The EKS Anywhere lifecycle controller enables you to use Kubernetes API-compatible clients such as `kubectl`, GitOps, or Terraform for managing workload clusters. In this EKS Anywhere version, the EKS Anywhere lifecycle controller rolls out new nodes in workload clusters when management clusters are upgraded.
+- In `eksctl anywhere` version `v0.16.0`, this behavior was changed such that management clusters can be upgraded separately from workload clusters.
```
[the remark this answers is not in the document]
Comment (csplinter) [reply docs/content/en/docs/clustermgmt/cluster-upgrades/baremetal-upgrades.md:23]: Good callout. We should always use "EKS Anywhere version" as opposed to "eksctl anywhere version", unless we are specifically referring to the version of the CLI

File: docs/content/en/docs/clustermgmt/cluster-upgrades/upgrade-overview.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+- Cluster components (Kubernetes, Cilium)
+
+You can find details about each EKS Anywhere releases in the EKS Anywhere release manifest. The release manifest contains references to the corresponding bundle manifest for each EKS Anywhere version. Within the bundle manifest, you will find the components included in a specific EKS Anywhere version. The images running in your deployment use the same URI values specified in the bundle manifest for that component. For example, see the [bundle manifest](https://anywhere-assets.eks.amazonaws.com/releases/bundles/59/manifest.yaml) for EKS Anywhere version `v0.18.7`.
+
+To upgrade the EKS Anywhere version of a management or standalone cluster, you install a new version of the `eksctl` CLI, change the `eksaVersion` field in your management or standalone cluster's spec yaml, and then run the `eksctl anywhere upgrade management-components -f cluster.yaml` (as of EKS Anywhere version v0.19) or `eksctl anywhere upgrade cluster -f cluster.yaml` command. The `eksctl anywhere upgrade cluster` command upgrades both management and cluster components.
```
Comment (abhinavmpandey08) [inline docs/content/en/docs/clustermgmt/cluster-upgrades/upgrade-overview.md:40]: nit 
```suggestion
To upgrade the EKS Anywhere version of a management or standalone cluster, you install a new version of the `eksctl anywhere` CLI, change the `eksaVersion` field in your management or standalone cluster's spec yaml, and then run the `eksctl anywhere upgrade management-components -f cluster.yaml` (as of EKS Anywhere version v0.19) or `eksctl anywhere upgrade cluster -f cluster.yaml` command. The `eksctl anywhere upgrade cluster` command upgrades both management and cluster components.
```

File: docs/content/en/docs/clustermgmt/cluster-upgrades/upgrade-overview.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+To upgrade the EKS Anywhere version of a workload cluster, you change the `eksaVersion` field in your workload cluster's spec yaml, and apply the new workload cluster's spec yaml to your management cluster using the `eksctl` CLI or with Kubernetes API-compatible clients.
+
+### Upgrading Kubernetes Versions
+
+Each EKS Anywhere version contains at least 4 minor versions of Kubernetes. Kubernetes patch version increments are included in EKS Anywhere minor and patch releases. There are two places in the cluster spec where you can configure the Kubernetes version, `Cluster.Spec.KubernetesVersion` and `Cluster.Spec.WorkerNodeGroupConfiguration[].KubernetesVersion`. If only `Cluster.Spec.KubernetesVersion` is set, then that version will apply to both control plane and worker nodes. You can use `Cluster.Spec.WorkerNodeGroupConfiguration[].KubernetesVersion` to upgrade your worker nodes separately from control plane nodes. The `Cluster.Spec.WorkerNodeGroupConfiguration[].KubernetesVersion` cannot be greater than `Cluster.Spec.KubernetesVersion`.
```
Comment (abhinavmpandey08) [inline docs/content/en/docs/clustermgmt/cluster-upgrades/upgrade-overview.md:46]: nit
```suggestion
Each EKS Anywhere version supports at least 4 minor versions of Kubernetes. Kubernetes patch version increments are included in EKS Anywhere minor and patch releases. There are two places in the cluster spec where you can configure the Kubernetes version, `Cluster.Spec.KubernetesVersion` and `Cluster.Spec.WorkerNodeGroupConfiguration[].KubernetesVersion`. If only `Cluster.Spec.KubernetesVersion` is set, then that version will apply to both control plane and worker nodes. You can use `Cluster.Spec.WorkerNodeGroupConfiguration[].KubernetesVersion` to upgrade your worker nodes separately from control plane nodes. The `Cluster.Spec.WorkerNodeGroupConfiguration[].KubernetesVersion` cannot be greater than `Cluster.Spec.KubernetesVersion`.
```

File: docs/content/en/docs/clustermgmt/cluster-upgrades/upgrade-overview.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+Reference the [EKS Anywhere Changelog]({{< relref "../../whatsnew/changelog/" >}}) for information on fixes, features, and changes included in each EKS Anywhere release. For details on the EKS Anywhere version support policy, reference the [Versioning page.]({{< relref "../../concepts/support-versions" >}})
+
+### Upgrade Version Skew
+
+{{% content "version-skew.md" %}}
+
+### User Interfaces
+
+Management and standalone clusters must be upgraded with the `eksctl` CLI. Workload clusters can be upgraded with the `eksctl` CLI or with Kubernetes API-compatible clients such as the `kubectl` CLI, GitOps, or Terraform. For an overview of the differences between management, standalone, workload clusters, reference the [Architecture page.]({{< relref "../../concepts/architecture" >}})
```
[the remark this answers is not in the document]
Comment (panktishah26) [reply docs/content/en/docs/clustermgmt/cluster-upgrades/upgrade-overview.md:22]: Yes, we can upgrade management cluster through controller/API.

[the remark this answers is not in the document]
Comment (csplinter) [reply docs/content/en/docs/clustermgmt/cluster-upgrades/upgrade-overview.md:22]: Ack, I will change to the following:

> EKS Anywhere versions for management and standalone clusters must be upgraded with the `eksctl anywhere` CLI. Kubernetes versions for management, standalone, and workload clusters, and EKS Anywhere versions for workload clusters can be upgraded with the `eksctl anywhere` CLI or with Kubernetes API-compatible clients such as the `kubectl` CLI, GitOps, or Terraform.

**Review by drewvanstone — COMMENTED** [review]
Couple small nits, but otherwise lgtm

**Review by drewvanstone — APPROVED** [review]

**Review by taneyland — APPROVED** [review]

**Pull request merged**